// @lowkey/react-native-gif/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.GifViewManager = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const GifViewManager = (0, _reactNative.requireNativeComponent)('GifImage');
exports.GifViewManager = GifViewManager;

const GifImage = props => {
  const {
    style,
    source,
    resizeMode,
    paused = false,
    onLoadEnd,
    useCPU = false,
    quality = 1
  } = props;

  const relovedSource = _reactNative.Image.resolveAssetSource(source);

  return _reactNative.Platform.OS === 'ios' ? /*#__PURE__*/_react.default.createElement(GifViewManager, {
    style: style,
    source: relovedSource.uri,
    resizeMode: resizeMode,
    paused: paused,
    onLoadEnd: onLoadEnd,
    useCPU: useCPU,
    quality: quality
  }) : /*#__PURE__*/_react.default.createElement(_reactNative.Image, props);
};

var _default = GifImage;
exports.default = _default;
